specs/packagemanager/PackageManagement.md

```diff
@@ -635,6 +644,66 @@ void CheckAndFixPackageVolume(string packageStorePath)
 }
 ```
 
+## 4.9. IsPackageProvisioned()
+
+Fabrikam app installing Contoso's Muffin and Waffle packages for all users if necessary, and with explicit user confirmation before the installation.
+
+```c#
+void Install()
+{
+    var packageSet = new PackageSet() {
+        Items = { new PackageSetItem() { PackageFamilyName = "contoso.muffin_1234567890abc",
+                                         MinVersion = ToVersion(1, 2, 3, 4),
+                                         PackageUri = new Uri("c:\\contoso\\muffin-1.2.3.4.msix") },
+                { new PackageSetItem() { PackageFamilyName = "contoso.waffle_1234567890abc",
+                                         MinVersion = ToVersion(2, 4, 6, 8),
+                                         PackageUri = new Uri("https://contoso.com/waffle-2.4.6.8.msix") } };
+
+    var packageDeploymentManager = PackageDeploymentManager.GetDefault();
+    if (!packageDeploymentManager.IsPackageSetProvisioned(packageSet))
```

Comment:
>IsPackageSetProvisioned [](http://example.com/codeflow?start=34&length=23)

why use a set?  Would it be better for caller to call in a loop then they can figure out which package is provisioned and which is not.

Reply:
Depends on your scenario. One I've seen is needing N packages provisioned and if 1+ aren't a simple user confirmation "I have stuff to do OK?" and if OK then do it, where details like package names aren't needed, desired or meaningful to the user.

Also, `PackagDeploymentManager` strives to offer consistent overloads across the various verbs accepting a package identity, a URI and a packageset. One of the reasons for PackageDeploymentManager's creation is PackageManager's consistently inconsistent API surface with corresponding usability pain points.

I don't see this is one of the more popular overloads across the API surface, but still non-zero.